contracts/zero-ex/contracts/src/features/ERC721OrdersFeature.sol

```diff
@@ -0,0 +1,1047 @@
+        _registerFeatureFunction(this.cancelERC721Order.selector);
+        _registerFeatureFunction(this.batchBuyERC721s.selector);
+        _registerFeatureFunction(this.matchERC721Orders.selector);
+        _registerFeatureFunction(this.batchMatchERC721Orders.selector);
+        _registerFeatureFunction(this.onERC721Received.selector);
+        _registerFeatureFunction(this.preSignERC721Order.selector);
+        _registerFeatureFunction(this.validateERC721OrderSignature.selector);
+        _registerFeatureFunction(this.validateERC721OrderProperties.selector);
+        _registerFeatureFunction(this.getERC721OrderStatus.selector);
+        _registerFeatureFunction(this.getERC721OrderHash.selector);
+        _registerFeatureFunction(this.getERC721OrderStatusBitVector.selector);
+        return LibMigrate.MIGRATE_SUCCESS;
+    }
+
+    /// @dev Sells an ERC721 asset to fill the given order.
+    /// @param order The ERC721 order.
+    /// @param signature The order signature from the maker.
+    /// @param erc721TokenId The ID of the ERC721 asset being
+    ///        sold. If the given order specifies properties, 
+    ///        the asset must satisfy those properties. Otherwise,
+    ///        it must equal the tokenId in the order. 
+    /// @param unwrapNativeToken If this parameter is true and the 
+    ///        ERC20 token of the order is e.g. WETH, unwraps the 
+    ///        token before transferring it to the taker.
+    function sellERC721(
+        LibERC721Order.ERC721Order memory order,
+        LibSignature.Signature memory signature,
+        uint256 erc721TokenId,
+        bool unwrapNativeToken
+    )
+        public
+        override
+    {
+        _sellERC721(
+            order, 
+            signature, 
+            erc721TokenId, 
+            unwrapNativeToken,
+            msg.sender, // taker
+            false       // isCallback
+        );
+    }
+
+    /// @dev Buys an ERC721 asset by filling the given order.
+    /// @param order The ERC721 order.
+    /// @param signature The order signature.
+    function buyERC721(
+        LibERC721Order.ERC721Order memory order,
+        LibSignature.Signature memory signature
+    )
+        public
+        override
+        payable
+    {
+        uint256 ethSpent = _buyERC721(order, signature, msg.value);
+        // Cannot spent more than `msg.value`
+        rrequire(
+            ethSpent <= msg.value,
+            LibERC721OrdersRichErrors.OverspentEthError(
+                ethSpent,
+                msg.value   
+            )
+        );
+        // Refund
+        if (ethSpent < msg.value) {
+            _transferEth(msg.sender, msg.value - ethSpent);
+        }
+    }
+
+    /// @dev Cancel a single ERC721 order. The caller must be the maker.
+    ///      Silently succeeds if the order has already been cancelled.
+    /// @param order The ERC721 order.
+    function cancelERC721Order(LibERC721Order.ERC721Order memory order)
```

Comment:
maybe change this function signature to just:
`function cancelERC721Order(address maker, uint256 nonce)`
for gas efficiency?

Reply:
(I did)